Tests/IntegrationTests/DropIn Tests/DropInTests.swift

```diff
@@ -194,6 +194,139 @@ class DropInTests: XCTestCase {
+        var config = DropInComponent.Configuration()
+        config.allowPreselectedPaymentView = false
+
+        var paymentMethods = try JSONDecoder().decode(PaymentMethods.self, from: DropInTests.paymentMethods.data(using: .utf8)!)
+        let storedPaymentMethod = try AdyenCoder.decode(storedCreditCardDictionary) as StoredCardPaymentMethod
+        paymentMethods.stored = [storedPaymentMethod]
+        
+        let sut = DropInComponent(paymentMethods: paymentMethods,
+                                  context: Dummy.context,
+                                  configuration: config)
+        
+        let storedPaymentMethodsDelegate = SessionStoredPaymentMethodDelegateMock()
+        storedPaymentMethodsDelegate.onDisable = { storedPM, dropIn in
+            XCTAssertEqual(storedPM.identifier, storedPaymentMethod.identifier)
+            return false
+        }
+        sut.storedPaymentMethodsDelegate = storedPaymentMethodsDelegate
+        XCTAssertNotNil(sut.sessionAsStoredPaymentMethodsDelegate)
+        
+        let expectation = expectation(description: "deletion delegate should be called")
+        let paymentMethodsListComponent = sut.paymentMethodListComponent(onCancel: nil)
+
+        sut.didDelete(storedPaymentMethod, in: paymentMethodsListComponent) { succes in
+            XCTAssertEqual(storedPaymentMethodsDelegate.onDisableCallCount, 1)
+            XCTAssertFalse(succes)
+            XCTAssertFalse(sut.paymentMethods.stored.isEmpty)
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 5)
+    }
+    
+    func testDeletingStoredPaymentSuccessAdvanced() throws {
+        var config = DropInComponent.Configuration()
+        config.allowPreselectedPaymentView = false
+
+        var paymentMethods = try JSONDecoder().decode(PaymentMethods.self, from: DropInTests.paymentMethods.data(using: .utf8)!)
+        let storedPaymentMethod = try AdyenCoder.decode(storedCreditCardDictionary) as StoredCardPaymentMethod
+        paymentMethods.stored = [storedPaymentMethod]
+        
+        let sut = DropInComponent(paymentMethods: paymentMethods,
+                                  context: Dummy.context,
+                                  configuration: config)
+        
+        let storedPaymentMethodsDelegate = StoredPaymentMethodDelegateMock()
+        storedPaymentMethodsDelegate.onDisable = { storedPM in
+            XCTAssertEqual(storedPM.identifier, storedPaymentMethod.identifier)
+            return true
+        }
+        sut.storedPaymentMethodsDelegate = storedPaymentMethodsDelegate
+        XCTAssertNil(sut.sessionAsStoredPaymentMethodsDelegate)
+        
+        let expectation = expectation(description: "deletion delegate should be called")
+        let paymentMethodsListComponent = sut.paymentMethodListComponent(onCancel: nil)
+
+        sut.didDelete(storedPaymentMethod, in: paymentMethodsListComponent) { succes in
```

Comment:
I was about to suggest to test error handling but interestingly we don't have any error to test according to the didDelete interface :)

Reply:
Yes mainly due to the api call for this being an empty response.